src/ducks/budgetAlerts/index.js

```diff
@@ -0,0 +1,106 @@
+import logger from 'cozy-logger'
+import isEqual from 'lodash/isEqual'
+import { TRANSACTION_DOCTYPE } from 'doctypes'
+import { getCategoryId } from 'ducks/categories/helpers'
+import sumBy from 'lodash/sumBy'
+import { fetchCategoryAlerts } from 'ducks/settings/helpers'
+
+const log = logger.namespace('category-alerts')
+
+const copyAlert = alert => ({ ...alert })
+
+/**
+ * Fetches transactions of current month corresponding to the alert
+ * Computes sum and returns information that used to send the global
+ * alert email
+ *
+ * Bails out if
+ *
+ * - sum is inferior to alert amount
+ * - the last reported amount inside alert is the same as the computed amount
+ *
+ * @param {Boolean} options.force Bypass last report checks
+ *
+ * @return {CategoryBudgetAlert}  - Updated alert (lastNotificationDate, lastNotificationAmount)
+ */
+const collectAlertInfo = async (client, alert, options) => {
+  const { data: monthExpenses } = await client.query(
+    client.all(TRANSACTION_DOCTYPE).where({
+      date: {
+        $lt: '2019-08',
+        $gt: '2019-07'
+      },
+      amount: {
+        $gt: 0
+      }
+    })
+  )
+  const categoryExpenses = monthExpenses.filter(
+    tr => getCategoryId(tr) === alert.categoryId
+  )
+  const sum = sumBy(categoryExpenses, tr => tr.amount)
+
+  if (sum < alert.balanceMinThreshold) {
```

Comment:
Nit: in my mind, we are not dealing with balances, so I would name it just `minThreshold`

Reply:
Changed to maxThreshold.